### PR TITLE
Deprecate gateway name/namespace from config file + internal packages

### LINKED ISSUE #3862: Deprecate gateway name/namespace from config file + internal packages
The Gateway Name + Namespace have been deprecated in favor of the GatewayClass.ControllerName field. When #3858 merges, ControllerName is actually used to look up the Gateway, so these fields can be fully deprecated and removed.

### PR DESCRIPTION
**Pull request by stevesloka**
Fixes #3862

Signed-off-by: Steve Sloka <slokas@vmware.com>

### CODE REVIEW

**Comment by stevesloka** [comment]
Need to add tests in https://github.com/projectcontour/contour/issues/3865 to validate the right Gateway is selected.

File: internal/dag/gatewayapi_processor.go
Comment (sunjayBhatia) [inline internal/dag/gatewayapi_processor.go:402]: maybe we should just bail out of the processor early and log if there is no gateway present so we don't have to keep doing these checks, it is invalid to generate config if there is no gateway anyways

**Review by sunjayBhatia — CHANGES_REQUESTED** [review]

**Comment by sunjayBhatia** [comment]
This change should also have an accompanying Operator change to remove the bits that set a Gateway name/namespace in the the Contour ConfigMap

File: internal/dag/gatewayapi_processor.go
Comment (skriss) [reply internal/dag/gatewayapi_processor.go:402]: Looks like we're already doing this (https://github.com/projectcontour/contour/blob/abeefb3f941951858b8b9f11a8775d896aba7757/internal/dag/gatewayapi_processor.go#L73-L81) so yeah we can drop these redundant nil checks

**Comment by sunjayBhatia** [comment]
Same as the linked operator issue:

Now that I think about it, might be required to move this to 1.19 so we don't re-break the operator/contour

right now it works ok having multiple contours b/c we're still passing the gateway name/namespace so a contour reconciles a single gateway

**Comment by skriss** [comment]
> Same as the linked operator issue:
> 
> Now that I think about it, might be required to move this to 1.19 so we don't re-break the operator/contour
> 
> right now it works ok having multiple contours b/c we're still passing the gateway name/namespace so a contour reconciles a single gateway

Yep seems like it'd be better to hold this until 1.19 unless there's a strong need to get it into 1.18.

File: pkg/config/parameters.go
Comment (mike1808) [inline pkg/config/parameters.go:58]: this should have been removed when the deprecation notice was added

Comment (stevesloka) [reply pkg/config/parameters.go:58]: We ended up going slower than I anticipated on this unfortunately. This will be cleaned up in Contour v1.19.

File: internal/dag/gatewayapi_processor.go
Comment (stevesloka) [reply internal/dag/gatewayapi_processor.go:402]: yup good call, I dropped this one in the method and kept the global one.

**Comment by sunjayBhatia** [comment]
made a draft here in the operator https://github.com/projectcontour/contour-operator/pull/422

removes the gatewayclass/gateway controllers from there so should make the operator compatible with this change

File: internal/dag/gatewayapi_processor.go
Comment (youngnick) [inline internal/dag/gatewayapi_processor.go:405]: I guess the linter is saying that this should be a `var` declaration instead?
```suggestion
		var gatewayNSName types.NamespacedName
```

File: internal/dag/gatewayapi_processor.go
Comment (youngnick) [inline internal/dag/gatewayapi_processor.go:439]: Probably better to have this one a `var` assignement as well, I guess.
```suggestion
	var gatewayNSName types.NamespacedName
```

File: internal/k8s/objectmeta.go
Comment (youngnick) [inline internal/k8s/objectmeta.go:25]: Nit: ?

**Review by youngnick — APPROVED** [review]
LGTM with a few small nits.

**Review by sunjayBhatia — APPROVED** [review]
LGTM, CI/linter not happy but thats an easy fix

Should fast-follow with operator PRs to keep CI healthy once this is merged

**Pull request merged**
